src/AppInstallerCLICore/Workflows/ShellExecuteInstallerHandler.cpp

```diff
@@ -307,7 +347,142 @@ namespace AppInstaller::CLI::Workflow
+                Utility::LocIndView locIndFeatureName{ featureName };
+
+                // Check if Windows Feature exists.
+#ifndef AICLI_DISABLE_TEST_HOOKS
+                auto doesFeatureExistResult = s_DoesWindowsFeatureExistResult_Override.has_value() ?
+                    s_DoesWindowsFeatureExistResult_Override.value() :
+                    DoesWindowsFeatureExist(context, featureName);
+#else
+                auto doesFeatureExistResult = DoesWindowsFeatureExist(context, featureName);
+#endif
 
+                if (!doesFeatureExistResult)
+                {
+                    // User aborted the operation.
+                    AICLI_TERMINATE_CONTEXT(E_ABORT);
+                }
+
+                auto doesFeatureExistExitCode = doesFeatureExistResult.value();
+                if (doesFeatureExistExitCode != ERROR_SUCCESS)
+                {
+                    AICLI_LOG(Core, Info, << "Windows Feature [" << featureName << "] does not exist");
+                    context.Reporter.Warn() << Resource::String::WindowsFeatureNotFound(locIndFeatureName) << std::endl;
+                    result = doesFeatureExistExitCode;
+                    return;
+                }
+
+                // Enable Windows Feature.
+                context.Reporter.Info() << Resource::String::EnablingWindowsFeature(locIndFeatureName) << std::endl;
+
+#ifndef AICLI_DISABLE_TEST_HOOKS
+                auto enableFeatureResult = s_EnableWindowsFeatureResult_Override.has_value() ?
+                    s_EnableWindowsFeatureResult_Override.value() :
+                    EnableWindowsFeature(context, featureName);
+#else
+                auto enableFeatureResult = EnableWindowsFeature(context, featureName);
+#endif
+                if (!enableFeatureResult)
+                {
+                    // User aborted the operation.
+                    AICLI_TERMINATE_CONTEXT(E_ABORT);
+                }
+
+                auto enableFeatureExitCode = enableFeatureResult.value();
+                if (enableFeatureExitCode == ERROR_SUCCESS_REBOOT_REQUIRED)
+                {
+                    AICLI_LOG(Core, Info, << "Reboot required for [" << featureName << "] with exit code: " << result);
+                    rebootRequired = true;
+                }
+                else if (FAILED(enableFeatureExitCode))
+                {
+                    AICLI_LOG(Core, Error, << "Failed to enable Windows Feature [" << featureName << "] with exit code: " << enableFeatureExitCode);
+                    context.Reporter.Warn() << Resource::String::FailedToEnableWindowsFeature(locIndFeatureName) << std::endl
+                        << GetUserPresentableMessage(enableFeatureExitCode) << std::endl;
+                    result = enableFeatureExitCode;
+                }
+                else
+                {
+                    AICLI_LOG(Core, Info, << "Successfully enabled [" << featureName << "] with exit code: " << result);

```

Comment:
```suggestion
                    AICLI_LOG(Core, Info, << "Successfully enabled [" << featureName << ']');

```
The result is not assigned, it's the value of last run